proposals/reports.md

```diff
@@ -0,0 +1,101 @@
+# Meta
+
+- Name: reports-v2
+- Start Date: 2022-09-12
+- Author(s): eddycharly
+- Supersedes: N/A
+
+# Table of Contents
+
+- [Meta](#meta)
+- [Table of Contents](#table-of-contents)
+- [Overview](#overview)
+- [Definitions](#definitions)
+- [Motivation](#motivation)
+- [Proposal](#proposal)
+- [Implementation](#implementation)
+- [Migration (OPTIONAL)](#migration-optional)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Unresolved Questions](#unresolved-questions)
+- [CRD Changes (OPTIONAL)](#crd-changes-optional)
+
+# Overview
+
+Support per resource report, automatic reports cleanup, and separate reports aggregation controller.
+
+# Definitions
+
+- TTL: Time-to-live. The amount of time a resource may exist until it is cleaned up.
+
+# Motivation
+
+The current implementation to generate and maintain policy reports is causing memory issues.
+Moreover, reports lifecycle are all managed by hand. Matching reports and their corresponding resource/policy is cumbersome.
+
+Kubernes has built-in mecanisms to clean up resources when parent resources are deleted, we should leverage native capabilities when possible.
+
+Processing reports should not impact Kyverno admission requests processing and we should be able to scale differently for large clusters when necessary.
+
+# Proposal
+
+In this proposal, we study the possibility to change the way reports are generated by:
+- creating one report per resource
+- bind the report lifecycle to the resource lifecycle
+- allow reconciling reports in an external process
+
+There are three ways of generating reports in Kyverno:
+1. At admission time, all policies running in audit mode are run against the admission request and produce report results.
+1. When a policy is created/updated/deleted, if the policy can run in background mode, reports are updated according to the policy changes.
+1. Periodically, policies running in background mode are re eveluated against resources present in the cluster and reports are updated accordingly.
```

Comment:
Index:
1.
2.
3.

Reply:
This will appear correctly in md viewer.